python/shotgun_model/shotgun_query_model.py

```diff
@@ -260,8 +264,22 @@ def hard_refresh(self):
 
         # delete cache file
         self._data_handler.remove_cache()
-        # request a reload
-        self._refresh_data()
+
+        # Clear ourselves, preserving the data handler so that we can then
+        # refresh. Clearing the model here ensures we don't end up with
+        # duplicate items once the cache is cleared and repopulated.
+        #
+        # Block all signals before we clear the model otherwise downstream
+        # proxy objects could cause crashes.
+        signals_blocked = self.blockSignals(True)
+        try:
+            # Clear all internal memory storage.
+            self.clear(remove_data_handler=False)
```

Comment:
i think of the `hard_refresh` almost as a helper method -- we want to clear caches on disk, and reset the model so that next time we load data, we start fresh. It seems to me that now we have a 'special' contract, meaning that in order for this to work we have to pass a special `remove_data_handler` flag to the clear method to 'clear the model in a special way'. It's not clear to me why! Do we need a special way to clear it? Would just `self.clear()` work?

Reply:
This has been refactored to take the small amount of work needed from clear() and use it directly instead of trying to bend clear() to my will. ;)

Hopefully it'll look more like what you're hoping for now!